File: VB/VB/E968.Module/Tests/TestController.vb
#If EasyTest Then
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.ExpressApp.SystemModule

Namespace DatabaseUserSettings.Tests
    Public Class TestController
        Inherits ViewController

        Public Sub New()
            Dim testAction As New SimpleAction(Me, "TestAction", DevExpress.Persistent.Base.PredefinedCategory.Tools)
            AddHandler testAction.Execute, AddressOf testAction_Execute
        End Sub
        Private Sub testAction_Execute(ByVal sender As Object, ByVal e As SimpleActionExecuteEventArgs)
            Dim navigationItems As IModelRootNavigationItems = CType(Application.Model, IModelApplicationNavigationItems).NavigationItems
            If SecuritySystem.CurrentUserName = DatabaseUserSettingsModule.ConfiguratorUserName Then
                navigationItems.StartupNavigationItem = navigationItems.AllItems("SecuritySystemRole_ListView")
            End If
            If SecuritySystem.CurrentUserName = "Sam" Then
                navigationItems.StartupNavigationItem = navigationItems.AllItems("SecuritySystemUser_ListView")
            End If
            Application.SaveModelChanges()
        End Sub
    End Class
End Namespace
#End If

File: CS/CS/E968.Module/Tests/TestController.cs
#if EasyTest
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.ExpressApp.SystemModule;

namespace DatabaseUserSettings.Tests {
    public class TestController : ViewController {
        public TestController() {
            SimpleAction testAction = new SimpleAction(this, "TestAction", DevExpress.Persistent.Base.PredefinedCategory.View);
            testAction.Execute += new SimpleActionExecuteEventHandler(testAction_Execute);
        }
        void testAction_Execute(object sender, SimpleActionExecuteEventArgs e) {
            IModelRootNavigationItems navigationItems = ((IModelApplicationNavigationItems)Application.Model).NavigationItems;
            if (SecuritySystem.CurrentUserName == DatabaseUserSettingsModule.ConfiguratorUserName)
                navigationItems.StartupNavigationItem = navigationItems.AllItems[2];
            if (SecuritySystem.CurrentUserName == "Sam")
                navigationItems.StartupNavigationItem = navigationItems.AllItems[3];
            Application.SaveModelChanges();
        }
    }
}
#endif
